pyvista/plotting/colors.py

```diff
@@ -945,36 +975,40 @@ def string_to_rgb(string):  # pragma: no cover
 
 def get_cmap_safe(cmap):
     """Fetch a colormap by name from matplotlib, colorcet, or cmocean."""
-    try:
-        from matplotlib.cm import get_cmap
-    except ImportError:  # pragma: no cover
-        raise ImportError(
-            'The use of custom colormaps requires the installation of matplotlib'
-        ) from None
-
     if isinstance(cmap, str):
         # check if this colormap has been mapped between ipygany
         if cmap in IPYGANY_MAP:
             cmap = IPYGANY_MAP[cmap]
 
         # Try colorcet first
-        try:
+        if _has_colorcet():
             import colorcet
 
-            cmap = colorcet.cm[cmap]
-        except (ImportError, KeyError):
-            pass
-        else:
-            return cmap
+            try:
+                cmap = colorcet.cm[cmap]
+            except KeyError:
+                pass
+            else:
+                return cmap
+
         # Try cmocean second
-        try:
+        if _has_cmocean():
             import cmocean
 
-            cmap = getattr(cmocean.cm, cmap)
-        except (ImportError, AttributeError):
-            pass
-        else:
-            return cmap
+            try:
+                cmap = getattr(cmocean.cm, cmap)
+            except AttributeError:
+                pass
+            else:
+                return cmap
+
+        try:
+            from matplotlib.cm import get_cmap
```

Comment:
I think it would be better to use `_has_cmap` here as well.

Reply:
I think if `matplotlib.cm` fails then `matplotlib` fails too, so this could be `_has_matplotlib`. Good catch.

Reply:
Implemented in 730f4a548c74d5aa5778b8d7cdc5a9ae3f1a2a92.